src/main/java/teammates/ui/request/CreateNotificationRequest.java

```diff
@@ -1,13 +1,16 @@
 package teammates.ui.request;
 
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.NotificationType;
+
 /**
  * The request for creating a notification.
  */
 public class CreateNotificationRequest extends BasicRequest {
     private Long startTimestamp;
     private Long endTimestamp;
-    private String notificationType;
-    private String targetUser;
+    private NotificationType notificationType;
+    private NotificationTargetUser targetUser;
```

Comment:
@t-cheepeng We might need some inputs from senior on how to validate enums. Currently it is validated when we get values for request parameters. If the value does not match any NotificationType, it will be null and will be caught by validation.

Reply:
Your current way of validating is fine, but in `NotificationAttributes` you are going to validate again (for the other fields) before inserting the data into db, so why not do all the validation there instead